internal/manifest/crypto.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2024 Edgeless Systems GmbH
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package manifest
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"fmt"
+
+	"github.com/edgelesssys/contrast/internal/userapi"
+)
+
+// MarshalSeedShareOwnerKey converts a public key into the format for userapi.SetManifestRequest.
+func MarshalSeedShareOwnerKey(pubKey *rsa.PublicKey) HexString {
+	return NewHexString(x509.MarshalPKCS1PublicKey(pubKey))
+}
+
+// ParseSeedShareOwnerKey reads a public key embedded in a userapi.SetManifestRequest.
+func ParseSeedShareOwnerKey(pubKeyHex HexString) (*rsa.PublicKey, error) {
```

Comment:
UnmarshalSeed... ?

Reply:
It's called `Marshal...` and `Parse...` in the `x509` package, though. Which inconsistency would you rather have?